lib/core/include/scipp/core/strides.h

```diff
@@ -31,27 +33,25 @@ class SCIPP_CORE_EXPORT Strides {
   bool operator==(const Strides &other) const noexcept;
   bool operator!=(const Strides &other) const noexcept;
 
-  constexpr scipp::index operator[](const scipp::index i) const {
-    return m_strides.at(i);
-  }
+  void emplace_back(const scipp::index i);
```

Comment:
```suggestion
  void push_back(const scipp::index i);
```
in keeping with the standard library, `emplace_back` would be a template and create an element by forwarding arguments to its constructor instead of taking a preconstructed element. (Makes no real difference here because we store integers, but still...)